Makefile

```diff
@@ -33,31 +33,41 @@ build: kill
-	$(DOCKER_COMPOSE_CMD) run collections-publisher bundle exec rake db:setup
-	$(DOCKER_COMPOSE_CMD) run contacts-admin bundle exec rake db:setup finders:publish
-	$(DOCKER_COMPOSE_CMD) run publisher bundle exec rake db:setup
-	$(DOCKER_COMPOSE_CMD) run frontend bundle exec rake publishing_api:publish_special_routes
-	$(DOCKER_COMPOSE_CMD) run content-tagger bundle exec rake db:setup
-	$(DOCKER_COMPOSE_CMD) run publishing-e2e-tests bundle exec rake wait_for_router
+	$(DOCKER_COMPOSE_CMD) run --rm --no-deps publishing-e2e-tests bash -c 'find /app/tmp -name .keep -prune -o -type f -exec rm {} \;'
+	$(DOCKER_COMPOSE_CMD) run --rm router-api bundle exec rake db:purge
+	$(DOCKER_COMPOSE_CMD) run --rm draft-router-api bundle exec rake db:purge
+	$(DOCKER_COMPOSE_CMD) run --rm content-store bundle exec rake db:purge
+	$(DOCKER_COMPOSE_CMD) run --rm draft-content-store bundle exec rake db:purge
+	$(DOCKER_COMPOSE_CMD) run --rm asset-manager bundle exec rake db:purge
+	$(DOCKER_COMPOSE_CMD) run --rm publishing-api bundle exec rake db:setup
+	$(DOCKER_COMPOSE_CMD) run --rm publishing-api bundle exec rake setup_exchange
+	$(DOCKER_COMPOSE_CMD) run --rm rummager-worker bundle exec rake message_queue:create_queues
+	$(DOCKER_COMPOSE_CMD) run --rm -e RUMMAGER_INDEX=all rummager bundle exec rake rummager:create_all_indices
+	$(DOCKER_COMPOSE_CMD) run --rm publishing-api-worker rails runner 'Sidekiq::Queue.new.clear'
+	$(DOCKER_COMPOSE_CMD) run --rm whitehall-admin bundle exec rake db:create db:purge db:setup publishing_api:publish_special_routes
+	$(DOCKER_COMPOSE_CMD) run --rm -e RUN_SEEDS_IN_PRODUCTION=true specialist-publisher bundle exec rake db:seed
+	$(DOCKER_COMPOSE_CMD) run --rm specialist-publisher bundle exec rake publishing_api:publish_finders
+	$(DOCKER_COMPOSE_CMD) run --rm travel-advice-publisher bundle exec rake db:seed
+	$(DOCKER_COMPOSE_CMD) run --rm manuals-publisher bundle exec rake db:seed
+	$(DOCKER_COMPOSE_CMD) run --rm collections-publisher bundle exec rake db:setup
+	$(DOCKER_COMPOSE_CMD) run --rm contacts-admin bundle exec rake db:setup finders:publish
+	$(DOCKER_COMPOSE_CMD) run --rm publisher bundle exec rake db:setup
+	$(DOCKER_COMPOSE_CMD) run --rm frontend bundle exec rake publishing_api:publish_special_routes
+	$(DOCKER_COMPOSE_CMD) run --rm content-tagger bundle exec rake db:setup
+	$(DOCKER_COMPOSE_CMD) run --rm publishing-e2e-tests bundle exec rake wait_for_router
 
 up:
 	$(DOCKER_COMPOSE_CMD) up -d
+	$(DOCKER_COMPOSE_CMD) run --rm publishing-e2e-tests \
+	./wait-for-it.sh www.dev.gov.uk:80 -- \
+	./wait-for-it.sh draft-origin.dev.gov.uk:80 -- \
+	./wait-for-it.sh router-api.dev.gov.uk:80 -- \
+	./wait-for-it.sh draft-router-api.dev.gov.uk:80 -- \
+	./wait-for-it.sh content-store.dev.gov.uk:80 -- \
+	./wait-for-it.sh draft-content-store.dev.gov.uk:80 -- \
+	./wait-for-it.sh publishing-api.dev.gov.uk:80 -- \
+	./wait-for-it.sh rummager.dev.gov.uk:80 -- \
+	./wait-for-it.sh asset-manager.dev.gov.uk:80
```

Comment:
Does it just test that it can make a tcp connection to the port? I'm just wondering if these actually verify any health as they could be bad gateways while app is starting up?

Reply:
Yes, it's unable to take the extra argument of the healthcheck routes and only supports a port